File: src/main/resources/static/icons/generate_icons.py
#!/usr/bin/env python3
"""
PWA 알약 아이콘 생성 스크립트
필요한 패키지: pip install Pillow
"""

from PIL import Image, ImageDraw
import os

def draw_pill_icon(size):
    """알약 캡슐 아이콘을 생성합니다."""
    # RGBA 모드로 이미지 생성 (투명 배경)
    image = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(image)

    # 캡슐 크기 계산
    center_x = size // 2
    center_y = size // 2
    pill_width = int(size * 0.7)
    pill_height = int(size * 0.4)
    radius = pill_height // 2

    # 캡슐 왼쪽 (파란색 반원)
    left_x = center_x - pill_width // 2
    blue_bbox = [
        left_x,
        center_y - radius,
        left_x + radius * 2,
        center_y + radius
    ]
    draw.ellipse(blue_bbox, fill=(59, 130, 246, 255))  # Blue

    # 캡슐 오른쪽 (빨간색 반원)
    right_x = center_x + pill_width // 2 - radius * 2
    red_bbox = [
        right_x,
        center_y - radius,
        right_x + radius * 2,
        center_y + radius
    ]
    draw.ellipse(red_bbox, fill=(239, 68, 68, 255))  # Red

    # 중앙 사각형 (왼쪽 파란색)
    blue_rect = [
        left_x + radius,
        center_y - radius,
        center_x,
        center_y + radius
    ]
    draw.rectangle(blue_rect, fill=(59, 130, 246, 255))

    # 중앙 사각형 (오른쪽 빨간색)
    red_rect = [
        center_x,
        center_y - radius,
        right_x + radius,
        center_y + radius
    ]
    draw.rectangle(red_rect, fill=(239, 68, 68, 255))

    # 중앙 분할선
    line_width = max(2, size // 100)
    draw.line(
        [(center_x, center_y - radius), (center_x, center_y + radius)],
        fill=(0, 0, 0, 100),
        width=line_width
    )

    # 테두리 추가
    outline_width = max(1, size // 100)

    # 왼쪽 반원 테두리
    draw.arc(blue_bbox, 90, 270, fill=(0, 0, 0, 50), width=outline_width)

    # 오른쪽 반원 테두리
    draw.arc(red_bbox, 270, 90, fill=(0, 0, 0, 50), width=outline_width)

    # 위쪽 선
    draw.line(
        [(left_x + radius, center_y - radius), (right_x + radius, center_y - radius)],
        fill=(0, 0, 0, 50),
        width=outline_width
    )

    # 아래쪽 선
    draw.line(
        [(left_x + radius, center_y + radius), (right_x + radius, center_y + radius)],
        fill=(0, 0, 0, 50),
        width=outline_width
    )

    return image

def main():
    """모든 크기의 아이콘을 생성합니다."""
    sizes = [72, 96, 128, 144, 152, 192, 384, 512]

    # 현재 스크립트 위치에 아이콘 저장
    script_dir = os.path.dirname(os.path.abspath(__file__))

    print("🎨 알약 아이콘 생성 시작...")

    for size in sizes:
        icon = draw_pill_icon(size)
        filename = f"icon-{size}x{size}.png"
        filepath = os.path.join(script_dir, filename)
        icon.save(filepath, 'PNG')
        print(f"✅ {filename} 생성 완료")

    print("\n🎉 모든 아이콘 생성 완료!")
    print(f"📁 저장 위치: {script_dir}")

if __name__ == "__main__":
    try:
        main()
    except ImportError:
        print("❌ Pillow 패키지가 설치되어 있지 않습니다.")
        print("다음 명령어로 설치하세요: pip install Pillow")
    except Exception as e:
        print(f"❌ 오류 발생: {e}")

File: src/main/resources/static/icons/generate_icons.js
#!/usr/bin/env node
/**
 * PWA 알약 아이콘 생성 스크립트 (Node.js)
 * 필요한 패키지: npm install canvas
 *
 * 사용법:
 * 1. npm install canvas
 * 2. node generate_icons.js
 */

const fs = require('fs');
const path = require('path');

// Canvas 라이브러리가 설치되어 있는지 확인
let Canvas;
try {
    Canvas = require('canvas');
} catch (err) {
    console.error('❌ canvas 패키지가 설치되어 있지 않습니다.');
    console.error('다음 명령어로 설치하세요: npm install canvas');
    process.exit(1);
}

const { createCanvas } = Canvas;

/**
 * 알약 캡슐 아이콘을 생성합니다.
 */
function drawPillIcon(size) {
    const canvas = createCanvas(size, size);
    const ctx = canvas.getContext('2d');

    // 배경 투명
    ctx.clearRect(0, 0, size, size);

    // 알약 캡슐 그리기
    const centerX = size / 2;
    const centerY = size / 2;
    const pillWidth = size * 0.7;
    const pillHeight = size * 0.4;
    const radius = pillHeight / 2;

    // 그림자
    ctx.shadowColor = 'rgba(0, 0, 0, 0.2)';
    ctx.shadowBlur = size * 0.05;
    ctx.shadowOffsetX = size * 0.02;
    ctx.shadowOffsetY = size * 0.02;

    // 캡슐 배경 (흰색 부분)
    ctx.fillStyle = '#FFFFFF';
    ctx.beginPath();
    ctx.arc(centerX - pillWidth/2 + radius, centerY, radius, Math.PI/2, Math.PI*3/2);
    ctx.arc(centerX + pillWidth/2 - radius, centerY, radius, -Math.PI/2, Math.PI/2);
    ctx.closePath();
    ctx.fill();

    // 그림자 제거
    ctx.shadowColor = 'transparent';

    // 캡슐 왼쪽 절반 (파란색)
    const gradient1 = ctx.createLinearGradient(
        centerX - pillWidth/2, centerY - radius,
        centerX - pillWidth/2, centerY + radius
    );
    gradient1.addColorStop(0, '#60A5FA');
    gradient1.addColorStop(1, '#3B82F6');

    ctx.fillStyle = gradient1;
    ctx.beginPath();
    ctx.arc(centerX - pillWidth/2 + radius, centerY, radius, Math.PI/2, Math.PI*3/2);
    ctx.lineTo(centerX, centerY - radius);
    ctx.lineTo(centerX, centerY + radius);
    ctx.closePath();
    ctx.fill();

    // 캡슐 오른쪽 절반 (빨간색)
    const gradient2 = ctx.createLinearGradient(
        centerX + pillWidth/2, centerY - radius,
        centerX + pillWidth/2, centerY + radius
    );
    gradient2.addColorStop(0, '#F87171');
    gradient2.addColorStop(1, '#EF4444');

    ctx.fillStyle = gradient2;
    ctx.beginPath();
    ctx.arc(centerX + pillWidth/2 - radius, centerY, radius, -Math.PI/2, Math.PI/2);
    ctx.lineTo(centerX, centerY + radius);
    ctx.lineTo(centerX, centerY - radius);
    ctx.closePath();
    ctx.fill();

    // 테두리
    ctx.strokeStyle = 'rgba(0, 0, 0, 0.1)';
    ctx.lineWidth = size * 0.01;
    ctx.beginPath();
    ctx.arc(centerX - pillWidth/2 + radius, centerY, radius, Math.PI/2, Math.PI*3/2);
    ctx.arc(centerX + pillWidth/2 - radius, centerY, radius, -Math.PI/2, Math.PI/2);
    ctx.closePath();
    ctx.stroke();

    // 중앙 분할선
    ctx.strokeStyle = 'rgba(0, 0, 0, 0.15)';
    ctx.lineWidth = size * 0.015;
    ctx.beginPath();
    ctx.moveTo(centerX, centerY - radius);
    ctx.lineTo(centerX, centerY + radius);
    ctx.stroke();

    // 하이라이트 효과
    ctx.fillStyle = 'rgba(255, 255, 255, 0.3)';
    ctx.beginPath();
    ctx.ellipse(
        centerX - pillWidth/4, centerY - radius/3,
        pillWidth/3, radius/2.5,
        0, 0, Math.PI * 2
    );
    ctx.fill();

    return canvas;
}

/**
 * 모든 크기의 아이콘을 생성합니다.
 */
function generateAllIcons() {
    const sizes = [72, 96, 128, 144, 152, 192, 384, 512];
    const outputDir = __dirname;

    console.log('🎨 알약 아이콘 생성 시작...\n');

    sizes.forEach(size => {
        const canvas = drawPillIcon(size);
        const filename = `icon-${size}x${size}.png`;
        const filepath = path.join(outputDir, filename);

        // PNG 버퍼로 변환하여 저장
        const buffer = canvas.toBuffer('image/png');
        fs.writeFileSync(filepath, buffer);

        console.log(`✅ ${filename} 생성 완료`);
    });

    console.log(`\n🎉 모든 아이콘 생성 완료!`);
    console.log(`📁 저장 위치: ${outputDir}`);
}

// 스크립트 실행
try {
    generateAllIcons();
} catch (error) {
    console.error('❌ 오류 발생:', error.message);
    process.exit(1);
}
